Install dependencies in Continuous Integration

Currently `.github/workflows/test.yaml` does not run because one of the dependencies that is imported `cgat` fails to correctly detect `numpy` even though this is explicitly installed before hand.

This needs fixing so that we can have Continuous Integration tests run on each merge to `main` to ensure changes to the code base do not break things.

For the minimal reproducible example in #62 has been merged so that refactoring can begin but we will need to fix this soon.

Made some progress, after a
[suggestion](https://github.com/cgat-developers/cgat-apps/issues/134#issuecomment-2497400890) I've added
`--no-build-isolation` to the step that installs `cgat` and/or `IsoSLAM`.

`cgat` is currently in a state of flux and as mentioned in the above linked thread the `v0.7.4` release was pulled from
PyPI. It included the build dependencies of `numpy` / `Cython` / `pysam` / `wheel` / `setuptools` in the `[build-system]
requires = [...]` of `pyproject.toml`.

This gets us a bit further but the external dependency check now fails. Firstly the external dependencies aren't
detected...


``` bash
Building wheels for collected packages: isoslam, cgat, cgatcore
  Building wheel for isoslam (pyproject.toml): started
  Building wheel for isoslam (pyproject.toml): finished with status 'done'
  Created wheel for isoslam: filename=isoslam-0.0.0-py3-none-any.whl size=13305 sha256=eb39d84b147ad37838bf94fe4d44ef8bfdf8404f83bda651ef9f4ea6aad2ede0
  Stored in directory: /tmp/pip-ephem-wheel-cache-3khbcrg7/wheels/ea/01/86/084cbcdf45bb92deb6358e1d26242db9d6a4b1c671067ea0b8
  Building wheel for cgat (setup.py): started
  Building wheel for cgat (setup.py): finished with status 'error'
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [460 lines of output]
      /tmp/pip-install-vv2ffijk/cgat_e93847922c304a74975b9256a7624f1c/setup.py:52: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
        if LooseVersion(setuptools.__version__) < LooseVersion('1.1'):
      WARNING: depency check for UCSC tools(wigToBigWig) failed, error 127
      WARNING: depency check for bedtools(bedtools) failed, error 127
      running bdist_wheel
```

...however the building of the C modules also fails...

``` bash
      writing manifest file 'cgat.egg-info/SOURCES.txt'
      /opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'cgat.BamTools' as data is deprecated, please list it in `packages`.
          !!


          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'cgat.BamTools' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.

          'cgat.BamTools' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).

          Please make sure that 'cgat.BamTools' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).

          You can read more about "package discovery" and "data files" on setuptools
          documentation page.


      !!

        check.warn(importable)
      /opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'cgat.Components' as data is deprecated, please list it in `packages`.
          !!


          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'cgat.Components' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.

          'cgat.Components' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).

          Please make sure that 'cgat.Components' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).

          You can read more about "package discovery" and "data files" on setuptools
          documentation page.


      !!

        check.warn(importable)
      /opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'cgat.FastqTools' as data is deprecated, please list it in `packages`.
          !!


          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'cgat.FastqTools' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.

          'cgat.FastqTools' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).

          Please make sure that 'cgat.FastqTools' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).

          You can read more about "package discovery" and "data files" on setuptools
          documentation page.


      !!

        check.warn(importable)
      /opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'cgat.VCFTools' as data is deprecated, please list it in `packages`.
          !!


          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'cgat.VCFTools' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.

          'cgat.VCFTools' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).

          Please make sure that 'cgat.VCFTools' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).

          You can read more about "package discovery" and "data files" on setuptools
          documentation page.


      !!

        check.warn(importable)
      copying cgat/GeneModelAnalysis.pyx -> build/lib.linux-x86_64-cpython-311/cgat
      creating build/lib.linux-x86_64-cpython-311/cgat/BamTools
      copying cgat/BamTools/bamtools.pyx -> build/lib.linux-x86_64-cpython-311/cgat/BamTools
      copying cgat/BamTools/geneprofile.pyx -> build/lib.linux-x86_64-cpython-311/cgat/BamTools
      copying cgat/BamTools/peakshape.pyx -> build/lib.linux-x86_64-cpython-311/cgat/BamTools
      creating build/lib.linux-x86_64-cpython-311/cgat/Components
      copying cgat/Components/Components.pyx -> build/lib.linux-x86_64-cpython-311/cgat/Components
      copying cgat/Components/connected_components.cpp -> build/lib.linux-x86_64-cpython-311/cgat/Components
      copying cgat/Components/connected_components.h -> build/lib.linux-x86_64-cpython-311/cgat/Components
      creating build/lib.linux-x86_64-cpython-311/cgat/FastqTools
      copying cgat/FastqTools/fastqtools.pyx -> build/lib.linux-x86_64-cpython-311/cgat/FastqTools
      creating build/lib.linux-x86_64-cpython-311/cgat/VCFTools
      copying cgat/VCFTools/vcftools.pyx -> build/lib.linux-x86_64-cpython-311/cgat/VCFTools
      copying cgat/NCL/_cnestedlist.pxd -> build/lib.linux-x86_64-cpython-311/cgat/NCL
      copying cgat/NCL/cnestedlist.pyx -> build/lib.linux-x86_64-cpython-311/cgat/NCL
      copying cgat/NCL/default.h -> build/lib.linux-x86_64-cpython-311/cgat/NCL
      copying cgat/NCL/intervaldb.c -> build/lib.linux-x86_64-cpython-311/cgat/NCL
      copying cgat/NCL/intervaldb.h -> build/lib.linux-x86_64-cpython-311/cgat/NCL
      running build_ext
      Compiling cgat/Components/Components.pyx because it changed.
      [1/1] Cythonizing cgat/Components/Components.pyx
      building 'cgat.Components' extension
      creating build/temp.linux-x86_64-cpython-311
      creating build/temp.linux-x86_64-cpython-311/cgat
      creating build/temp.linux-x86_64-cpython-311/cgat/Components
      gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11 -c cgat/Components/Components.cpp -o build/temp.linux-x86_64-cpython-311/cgat/Components/Components.o
      gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11 -c cgat/Components/connected_components.cpp -o build/temp.linux-x86_64-cpython-311/cgat/Components/connected_components.o
      g++ -shared -Wl,--rpath=/opt/hostedtoolcache/Python/3.11.10/x64/lib -Wl,--rpath=/opt/hostedtoolcache/Python/3.11.10/x64/lib build/temp.linux-x86_64-cpython-311/cgat/Components/Components.o build/temp.linux-x86_64-cpython-311/cgat/Components/connected_components.o -L/opt/hostedtoolcache/Python/3.11.10/x64/lib -o build/lib.linux-x86_64-cpython-311/cgat/Components.cpython-311-x86_64-linux-gnu.so
      Compiling cgat/NCL/cnestedlist.pyx because it changed.
      [1/1] Cythonizing cgat/NCL/cnestedlist.pyx
      warning: cgat/NCL/cnestedlist.pyx:106:24: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:106:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:389:24: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:389:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:265:26: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:265:30: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:353:26: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      warning: cgat/NCL/cnestedlist.pyx:353:30: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
      building 'cgat.NCL.cnestedlist' extension
      creating build/temp.linux-x86_64-cpython-311/cgat/NCL
      gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11 -c cgat/NCL/cnestedlist.c -o build/temp.linux-x86_64-cpython-311/cgat/NCL/cnestedlist.o
      gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11 -c cgat/NCL/intervaldb.c -o build/temp.linux-x86_64-cpython-311/cgat/NCL/intervaldb.o
      cgat/NCL/intervaldb.c: In function ‘read_imdiv’:
      cgat/NCL/intervaldb.c:486:3: warning: ignoring return value of ‘fread’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
        486 |   fread(imdiv,sizeof(IntervalMap),block,ifile);
            |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c: In function ‘read_sublist’:
      cgat/NCL/intervaldb.c:502:3: warning: ignoring return value of ‘fread’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
        502 |   fread(im,sizeof(IntervalMap),subheader->len,ifile);
            |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c: In function ‘read_subheader_block’:
      cgat/NCL/intervaldb.c:521:3: warning: ignoring return value of ‘fread’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
        521 |   fread(subheader,sizeof(SublistHeader),nblock,ifile);
            |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c: In function ‘read_binary_files’:
      cgat/NCL/intervaldb.c:829:3: warning: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
        829 |   fscanf(ifile,"%d %d %d %d %d",&n,&ntop,&div,&nlists,&nii);
            |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c:841:5: warning: ignoring return value of ‘fread’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
        841 |     fread(ii,sizeof(IntervalIndex),nii,ifile);
            |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c: In function ‘write_binary_files’:
      cgat/NCL/intervaldb.c:762:42: warning: ‘%s’ directive writing up to 2047 bytes into a region of size 1004 [-Wformat-overflow=]
        762 |     sprintf(err_msg,"unable to open file %s for writing",path);
            |                                          ^~              ~~~~
      In file included from /usr/include/stdio.h:894,
                       from /opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11/Python.h:24,
                       from cgat/NCL/default.h:13,
                       from cgat/NCL/intervaldb.h:4,
                       from cgat/NCL/intervaldb.c:2:
      /usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10: note: ‘__builtin___sprintf_chk’ output between 33 and 2080 bytes into a destination of size 1024
         38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         39 |                                   __glibc_objsize (__s), __fmt,
            |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         40 |                                   __va_arg_pack ());
            |                                   ~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c: In function ‘text_file_to_binaries’:
      cgat/NCL/intervaldb.c:1015:19: warning: ‘.size’ directive writing 5 bytes into a region of size between 1 and 2048 [-Wformat-overflow=]
       1015 |   sprintf(path,"%s.size",filestem); /* SAVE BASIC SIZE INFO*/
            |                   ^~~~~
      In file included from /usr/include/stdio.h:894,
                       from /opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11/Python.h:24,
                       from cgat/NCL/default.h:13,
                       from cgat/NCL/intervaldb.h:4,
                       from cgat/NCL/intervaldb.c:2:
      /usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10: note: ‘__builtin___sprintf_chk’ output between 6 and 2053 bytes into a destination of size 2048
         38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         39 |                                   __glibc_objsize (__s), __fmt,
            |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         40 |                                   __va_arg_pack ());
            |                                   ~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c:1029:21: warning: ‘.index’ directive writing 6 bytes into a region of size between 1 and 2048 [-Wformat-overflow=]
       1029 |     sprintf(path,"%s.index",filestem); /* SAVE INDEX INFO*/
            |                     ^~~~~~
      In file included from /usr/include/stdio.h:894,
                       from /opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11/Python.h:24,
                       from cgat/NCL/default.h:13,
                       from cgat/NCL/intervaldb.h:4,
                       from cgat/NCL/intervaldb.c:2:
      /usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10: note: ‘__builtin___sprintf_chk’ output between 7 and 2054 bytes into a destination of size 2048
         38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         39 |                                   __glibc_objsize (__s), __fmt,
            |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         40 |                                   __va_arg_pack ());
            |                                   ~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c:1045:21: warning: ‘.subhead’ directive writing 8 bytes into a region of size between 1 and 2048 [-Wformat-overflow=]
       1045 |     sprintf(path,"%s.subhead",filestem); /* SAVE THE SUBHEADER LIST */
            |                     ^~~~~~~~
      In file included from /usr/include/stdio.h:894,
                       from /opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11/Python.h:24,
                       from cgat/NCL/default.h:13,
                       from cgat/NCL/intervaldb.h:4,
                       from cgat/NCL/intervaldb.c:2:
      /usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10: note: ‘__builtin___sprintf_chk’ output between 9 and 2056 bytes into a destination of size 2048
         38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         39 |                                   __glibc_objsize (__s), __fmt,
            |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         40 |                                   __va_arg_pack ());
            |                                   ~~~~~~~~~~~~~~~~~
      cgat/NCL/intervaldb.c:1061:19: warning: ‘.idb’ directive writing 4 bytes into a region of size between 1 and 2048 [-Wformat-overflow=]
       1061 |   sprintf(path,"%s.idb",filestem); /* SAVE THE ACTUAL INTERVAL DB*/
            |                   ^~~~
      In file included from /usr/include/stdio.h:894,
                       from /opt/hostedtoolcache/Python/3.11.10/x64/include/python3.11/Python.h:24,
                       from cgat/NCL/default.h:13,
                       from cgat/NCL/intervaldb.h:4,
                       from cgat/NCL/intervaldb.c:2:
      /usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10: note: ‘__builtin___sprintf_chk’ output between 5 and 2052 bytes into a destination of size 2048
         38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
            |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         39 |                                   __glibc_objsize (__s), __fmt,
            |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         40 |                                   __va_arg_pack ());
            |                                   ~~~~~~~~~~~~~~~~~
      gcc -shared -Wl,--rpath=/opt/hostedtoolcache/Python/3.11.10/x64/lib -Wl,--rpath=/opt/hostedtoolcache/Python/3.11.10/x64/lib build/temp.linux-x86_64-cpython-311/cgat/NCL/cnestedlist.o build/temp.linux-x86_64-cpython-311/cgat/NCL/intervaldb.o -L/opt/hostedtoolcache/Python/3.11.10/x64/lib -o build/lib.linux-x86_64-cpython-311/cgat/NCL/cnestedlist.cpython-311-x86_64-linux-gnu.so
      Compiling cgat/GeneModelAnalysis.pyx because it changed.
      [1/1] Cythonizing cgat/GeneModelAnalysis.pyx

      Error compiling Cython file:
      ------------------------------------------------------------
      ...
                  utr5, utr3 = utr3, utr5
              return utr5, utr3

      cimport numpy
      DTYPE_INT = numpy.int64
      ctypedef numpy.int_t DTYPE_INT_t
               ^
      ------------------------------------------------------------

      cgat/GeneModelAnalysis.pyx:311:9: 'int_t' is not a type identifier
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-vv2ffijk/cgat_e93847922c304a74975b9256a7624f1c/setup.py", line 298, in <module>
          setup(
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
          self.run_command(cmd)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/wheel/_bdist_wheel.py", line 387, in run
          self.run_command("build")
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
          self.distribution.run_command(command)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 132, in run
          self.run_command(cmd_name)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
          self.distribution.run_command(command)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 346, in run
          self.build_extensions()
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 466, in build_extensions
          self._build_extensions_serial()
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 492, in _build_extensions_serial
          self.build_extension(ext)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/Cython/Distutils/build_ext.py", line 130, in build_extension
          new_ext = cythonize(
                    ^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 1154, in cythonize
          cythonize_one(*args)
        File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 1321, in cythonize_one
          raise CompileError(None, pyx_file)
      Cython.Compiler.Errors.CompileError: cgat/GeneModelAnalysis.pyx
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cgat
  Running setup.py clean for cgat
  Building wheel for cgatcore (setup.py): started
  Building wheel for cgatcore (setup.py): finished with status 'done'
  Created wheel for cgatcore: filename=cgatcore-0.6.16-py3-none-any.whl size=119476 sha256=e55b56915d6e53eb0caaabcd4dc18d8f9bb6d3ffa0db777a008bb0fd0ac3f377
  Stored in directory: /home/runner/.cache/pip/wheels/29/9b/6d/b62a62fcfeb6c07a516b2e5c79c7300b93b7c21e39a36021b8
Successfully built isoslam cgatcore
Failed to build cgat
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (cgat)
Error: Process completed with exit code 1.
```

It might be related to the deprecation warnings...

``` bash
      /opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'cgat.Components' as data is deprecated, please list it in `packages`.

```

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -27,8 +27,8 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        os: ["ubuntu-latest", "macos-latest"]
+        python-version: ["3.10", "3.11", "3.12"]
     steps:
       - uses: actions/checkout@v4
       - name: Set up Python
@@ -40,10 +40,6 @@ jobs:
           pip install --upgrade virtualenv
           pip install --upgrade pip setuptools
           virtualenv --upgrade-embed-wheels
-          pip install numpy
-          pip show numpy
-          pip install cgatcore
-          pip install cgat
           pip install -e .[tests]
       - name: Test with pytest
         run: |

--- a/isoslam/all_introns_counts_and_info.py
+++ b/isoslam/all_introns_counts_and_info.py
@@ -12,14 +12,10 @@ read, checks these are not present in the SNP VCF file, and outputs metadata on 
 about it's event assignment, number of conversions, coverage etc.
 """
 
+from argparse import ArgumentParser
 import sys
-from collections import defaultdict
 
-import cgat.GTF as GTF
-import cgatcore.experiment as E
-import cgatcore.iotools as iotools
 import pandas as pd
-import pysam as pysam
 
 from isoslam import isoslam, io
 
@@ -31,9 +27,7 @@ def main(argv=None):
     if argv is None:
         argv = sys.argv
 
-    # setup command line parser
-    parser = E.ArgumentParser(description=__doc__)
-
+    parser = ArgumentParser(description=__doc__)
     parser.add_argument(
         "-b",
         "--bam",
@@ -41,15 +35,12 @@ def main(argv=None):
         type=str,
         help="Supply a path to the bam file that has undergone read assignment with featureCounts",
     )
-
     parser.add_argument(
         "-g", "--gtf", dest="gtf_path", type=str, help="Supply a path to the transcript assembly gtf file"
     )
-
     parser.add_argument(
         "-bed", dest="utron_bed", type=str, help="Supply a path to the utron bed file. Must be bed6 format"
     )
-
     parser.add_argument(
         "-o",
         "--out",
@@ -58,7 +49,6 @@ def main(argv=None):
         help="""Supply a path to the output file. This file will contain
                         conversions per pair, accounting for stranding""",
     )
-
     parser.add_argument("-vcf", "--vcf", dest="vcf_path", type=str, help="""Supply a path to the VCF.gz file""")
 
     argv_as_dictionary = vars(argv)
@@ -118,8 +108,6 @@ def main(argv=None):
             i_progress += 1
 
             if i_progress == 10000:
-                # E.debug(str(i_total_progress) + " pairs processed")
-                # E.debug(str(i) + "spliced/retained pairs processed")
                 i_progress = 0
 
             read1_start = read1.reference_start
@@ -506,8 +494,6 @@ def main(argv=None):
                 results = pd.concat([results, row])
 
     return results.sort_values(by=["read_uid", "transcript_id", "chr", "start", "end"])
-    # write footer and output benchmark information.
-    # E.stop()
 
 
 if __name__ == "__main__":

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -36,8 +36,6 @@ keywords = [
 ]
 dependencies = [
   "apsw",
-  # "cgat",
-  # "cgatcore",
   "gevent",
   "loguru",
   "matplotlib",
@@ -45,7 +43,6 @@ dependencies = [
   "pandas",
   "pysam",
   "ruamel.yaml",
-  # "ruffus",
   "sqlalchemy",
 ]
 
@@ -57,6 +54,7 @@ tests = [
   "pytest-cov",
   "pytest-github-actions-annotate-failures",
   "pytest-mpl",
+  "pytest-regtest",
   "pytest-testmon",
   "syrupy",
 ]

